noxfile.py

```diff
@@ -35,3 +36,69 @@ def mypy(session: nox.Session):
 def test(session: nox.Session):
     session.install("pytest", ".")
     session.run("pytest", "setuptools_rust", *session.posargs)
+
+
+def download_pyodide(session: nox.Session, pyodide_dir: Path) -> None:
+    pyodide_dir.mkdir()
+
+    PYODIDE_DEV = "https://pyodide-cdn2.iodide.io/dev/full/"
+    pyodide_files = [
+        "pyodide.js",
+        "packages.json",
+        "pyodide.asm.js",
+        "pyodide.asm.data",
+        "pyodide.asm.wasm",
+        "pyodide_py.tar",
+    ]
+    with session.chdir(pyodide_dir):
+        for file in pyodide_files:
+            session.run("wget", PYODIDE_DEV + file, external=True)
+        session.run("npm", "i", "node-fetch", external=True)
+
+
+@nox.session(name="test-examples-emscripten")
+def test_examples_emscripten(session: nox.Session):
+    session.install(".")
+    emscripten_dir = Path("./emscripten").resolve()
+    pyodide_dir = emscripten_dir / "pyodide"
+    if not pyodide_dir.exists():
+        download_pyodide(session, pyodide_dir)
+
+    session.run(
+        "rustup",
+        "component",
+        "add",
+        "rust-src",
+        "--toolchain",
+        "nightly",
+        external=True,
+    )
+    for example in Path("examples/").resolve().glob("*"):
+        if example.name == "hello-world":
+            continue
+        env = os.environ.copy()
+        env.update(
+            RUSTUP_TOOLCHAIN="nightly",
+            _SETUPTOOLSRUST_BUILD_STD="1",
+            PYTHONPATH=str(emscripten_dir),
+            _PYTHON_SYSCONFIGDATA_NAME="_sysconfigdata__emscripten_wasm32-emscripten",
+            _PYTHON_HOST_PLATFORM="emscripten_3_1_13_wasm32",
+            CARGO_BUILD_TARGET="wasm32-unknown-emscripten",
+            CARGO_TARGET_WASM32_UNKNOWN_EMSCRIPTEN_LINKER=str(
+                emscripten_dir / "emcc_wrapper.py"
+            ),
+            PYO3_CONFIG_FILE=str(emscripten_dir / "pyo3_config.ini"),
+            RUSTFLAGS=" ".join(
```

Comment:
It might be worth having a .cargo/config.toml with a `[target.wasm32-unknown-emscripten]` section for these? Or alternatively `setuptools-rust` could set them automatically internally?

Reply:
Well we are working on upstreaming this stuff:
https://github.com/rust-lang/rust/pulls?q=author%3Ahoodmane
I don't know what the best thing to do before the upstream fixes go in. Unfortunately it looks like the merge window for Rust 1.63.0 is closing today. The fixes will probably go into into 1.64.0.

Reply:
Ah, very nice. I think it's ok for us to keep this as-is for now then; if things will eventually "just work" on a new enough rustc I think that's good enough for now.